### ISSUE TITLE
Could not initialize seastar: std::out_of_range (_Map_base::at)

### ISSUE THREAD

**Issue by MasterGroosha** [scylladb/scylla#3779]
- [*] I have read the disclaimer above, and I am reporting a suspected malfunction in Scylla.

*Installation details*
Scylla version (or git commit hash): 2.3.0-0.20180917.178f870a0
Cluster size: 1 host (standalone mode)
OS (RHEL/CentOS/Ubuntu/AWS AMI): Ubuntu 18.04.1

*Hardware details (for performance issues)* 
Platform (physical/VM/cloud instance type/docker): Physical
Disks: (SSD/HDD, count) 5 disks as RAID-0 (xfs) + 1 disk as ext4 for commit log

Fresh installation of ScyllaDB doesn't start with exception in title. Performed scylla_setup and scylla_io_setup.

`io.conf` file:
`SEASTAR_IO="--num-io-queues=1 --io-properties-file=/etc/scylla.d/io_properties.yaml"`

`io_properties.yaml` file:
```
disks:
  - mountpoint: /media/scylla/data # This directory exists and has 777 rights (for testing)
    read_iops: 826
    read_bandwidth: 519884480
    write_iops: 1058
    write_bandwidth: 618628672
```

part of `scylla.yaml` config file:
```
data_file_directories:
    - /media/scylla/data

commitlog_directory: /media/scylla-logs/commitlog # this dir also exists with the same rights
```

However when I run `systemctl start scylla-server`, I see the following error in journalctl:
```
Sep 25 08:07:56 server scylla[8235]: Could not initialize seastar: std::out_of_range (_Map_base::at)\
Sep 25 08:07:56 server systemd[1]: scylla-server.service: Main process exited, code=exited, status=1/FAILURE
Sep 25 08:07:56 server systemd[1]: scylla-server.service: Failed with result 'exit-code'.
Sep 25 08:07:56 server systemd[1]: Failed to start Scylla Server.
```

Unfortunately I don't see any other errors anywhere, there's no `scylla` directory in `/var/log`. `kern.log` and `syslog` show me no new information about scylla. Could you please assist me in troubleshooting this problem and finding a cause of it? 
Upd: looks like sometimes Scylla produces backtrace as shown in #3636, how can I see it in my case?

**Comment by MasterGroosha** [comment]
Upd: also related #2195 and https://github.com/scylladb/seastar/issues/265

**Comment by glommer** [comment]
I don't think that issue is related, because it is long closed.

One thing that would help is to do `journalctl -u scylla-server`, as you will see more messages.
Also, how many physical sockets do you have in your physical box?

**Comment by MasterGroosha** [comment]
@glommer I've shown pretty much everything from journalctl above. If only: (reversed)
```
-- Logs begin at Fri 2018-09-07 15:22:15 UTC, end at Tue 2018-09-25 13:35:01 UTC. --
сен 25 13:32:18 server systemd[1]: Failed to start Scylla Server.
сен 25 13:32:18 server systemd[1]: scylla-server.service: Failed with result 'exit-code'.
сен 25 13:32:18 server systemd[1]: scylla-server.service: Main process exited, code=exited, status=1/FAILURE
сен 25 13:32:18 server scylla[17065]: Could not initialize seastar: std::out_of_range (_Map_base::at)
сен 25 13:32:18 server scylla_prepare[17051]: tuning /sys/devices/pci0000:00/0000:00:02.0/0000:04:00.0/ata8/host9/target9:0:0/9:0:0:0/block/sdg/sdg2
сен 25 13:32:18 server scylla_prepare[17051]: tuning: /sys/devices/pci0000:00/0000:00:02.0/0000:04:00.0/ata8/host9/target9:0:0/9:0:0:0/block/sdg/queue/nomerges 2
сен 25 13:32:18 server scylla_prepare[17051]: tuning: /sys/devices/pci0000:00/0000:00:02.0/0000:04:00.0/ata8/host9/target9:0:0/9:0:0:0/block/sdg/queue/scheduler noop
сен 25 13:32:18 server scylla_prepare[17051]: tuning /sys/devices/pci0000:00/0000:00:02.0/0000:04:00.0/ata8/host9/target9:0:0/9:0:0:0/block/sdg
сен 25 13:32:18 server scylla_prepare[17051]: tuning /sys/devices/pci0000:00/0000:00:02.0/0000:04:00.0/ata8/host9/target9:0:0/9:0:0:0/block/sdg/sdg2
```

I have 4 sockets in server, and Scylla worked fine on previous install of Ubuntu 18 with 2.3 rc-2. However with 2.3 release and on fresh install I have this error.

**Comment by glommer** [comment]
can you try editing /etc/scylla.d/io.conf so it reads --num-io-queues=4 ?

**Comment by MasterGroosha** [comment]
Yes, it worked! Thanks! 
This was the last thing I was unsure about, because `sudo scylla_io_setup` explicitly told me

```
INFO  2018-09-25 14:06:22,395 [shard 0] iotune - /media/scylla/data passed sanity checks
Measuring sequential write bandwidth: 556 MB/s
Measuring sequential read bandwidth: 629 MB/s
Measuring random write IOPS: 1130 IOPS
Measuring random read IOPS: 753 IOPS
Recommended --num-io-queues: 1
```
(and it overwritten my io.conf with value 1 instead of 4)

**Comment by glommer** [comment]
There is a requirement to use one queue per socket.
The requirement was already there before, but maybe the old configuration for some reason got you the minimum amount of queues.

Please don't close this bug : `scylla_io_setup` should work out of the box.

**Comment by gibsn** [comment]
Experiencing the same issue

4 servers, Centos 7, 2 CPUs, 10 HDD RAID10, SSD RAID1

2 servers starting normally, 2 would not start with the same exception. Those 2 starting normally had enormous IOPS (like billions or more) in `scylla_io_setup`. Also since we are trying to configure server ourselves I did not run the full `scylla_setup` script.

As far as I understand invalid IOPS results for first 2 machines resulted in `--num-io-queues=32`. Not running `scylla_sysconfig_setup` resulted in `cpuset.conf` not being set properly which resulted in `--num-io-queues=1` on other machines.

**Comment by glommer** [comment]
For the purposes of the folks hitting this issue:

Scylla needs to see at least as many I/O Queues as there are sockets in the machine, which is what causes this issue. The fix I intend to deliver is to patch scylla_io_config to guarantee that as a minimum instead of one.

**Comment by glommer** [comment]
Hello @gibsn and @MasterGroosha 

What does `lscpu` say in your system ?

**Comment by MasterGroosha** [comment]
@glommer First of all, I should say I'm sorry for a bit of lie here: while investigating the issue with you it turned out it only had 2 sockets, not 4 (after I posted my comment about 4 sockets). Your advice still worked, since I changed `--num-io-queues` to 2 and Scylla started like a charm.

`lscpu` output:
```
Architecture:        x86_64
CPU op-mode(s):      32-bit, 64-bit
Byte Order:          Little Endian
CPU(s):              24
On-line CPU(s) list: 0-23
Thread(s) per core:  2
Core(s) per socket:  6
Socket(s):           2
NUMA node(s):        2
Vendor ID:           GenuineIntel
CPU family:          6
Model:               45
Model name:          Intel(R) Xeon(R) CPU E5-2640 0 @ 2.50GHz
Stepping:            7
```

And `scylla_io_setup` does change `--num-io-queues` value to 1 after benchmark.

**Comment by gibsn** [comment]
@glommer 
```
# lscpu
Architecture:          x86_64
CPU op-mode(s):        32-bit, 64-bit
Byte Order:            Little Endian
CPU(s):                32
On-line CPU(s) list:   0-31
Thread(s) per core:    2
Core(s) per socket:    8
Socket(s):             2
NUMA node(s):          2
Vendor ID:             GenuineIntel
CPU family:            6
Model:                 79
Model name:            Intel(R) Xeon(R) CPU E5-2620 v4 @ 2.10GHz
Stepping:              1
CPU MHz:               2101.000
CPU max MHz:           2101.0000
CPU min MHz:           1200.0000
BogoMIPS:              4190.40
Virtualization:        VT-x
L1d cache:             32K
L1i cache:             32K
L2 cache:              256K
L3 cache:              20480K
NUMA node0 CPU(s):     0-7,16-23
NUMA node1 CPU(s):     8-15,24-31
Flags:                 ...```

**Comment by bwangfdu** [comment]
I am facing this bug as well. Here is my lscpu result:
`root@stor4:~# lscpu
Architecture:        x86_64
CPU op-mode(s):      32-bit, 64-bit
Byte Order:          Little Endian
CPU(s):              16
On-line CPU(s) list: 0-15
Thread(s) per core:  2
Core(s) per socket:  8
Socket(s):           1
NUMA node(s):        4
Vendor ID:           AuthenticAMD
CPU family:          23
Model:               1
Model name:          AMD EPYC 7251 8-Core Processor
Stepping:            2
CPU MHz:             2894.413
BogoMIPS:            4192.02
Virtualization:      AMD-V
L1d cache:           32K
L1i cache:           64K
L2 cache:            512K
L3 cache:            4096K
NUMA node0 CPU(s):   0,4,8,12
NUMA node1 CPU(s):   1,5,9,13
NUMA node2 CPU(s):   2,6,10,14
NUMA node3 CPU(s):   3,7,11,15
Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl nonstop_tsc cpuid extd_apicid amd_dcm aperfmperf pni pclmulqdq monitor ssse3 fma cx16 sse4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw skinit wdt tce topoext perfctr_core perfctr_nb bpext perfctr_llc mwaitx cpb hw_pstate sme ssbd ibpb vmmcall fsgsbase bmi1 avx2 smep bmi2 rdseed adx smap clflushopt sha_ni xsaveopt xsavec xgetbv1 xsaves clzero irperf xsaveerptr arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold avic v_vmsave_vmload vgif overflow_recov succor smca
`
After io tuning I get --num-io-queues as 1, but I should change it to 4 to make the service start.

**Comment by glommer** [comment]
Thanks - my apologies that we didn't yet have the time to fix this

**Comment by slivne** [comment]
@glommer - did we fix this ?

**Comment by glommer** [comment]
unfortunately not

**Comment by jackivanov** [comment]
> Scylla needs to see at least as many I/O Queues as there are sockets in the machine, which is what causes this issue. The fix I intend to deliver is to patch scylla_io_config to guarantee that as a minimum instead of one.

Not to sockets, actually, but to NUMA nodes, right?

**Comment by glommer** [comment]
Right.

**Comment by davidt99** [comment]
Just to mention that we couldn't override `io.conf` docker.

**Comment by eliransin** [comment]
Fixed in version Scylla version 3.1 by commit b39eac653
